projects/igniteui-angular/src/lib/grids/grid-navigation.service.ts

```diff
@@ -456,15 +457,67 @@ export class IgxGridNavigationService {
         }
     }
 
-    public moveFocusToFilterCell() {
+    public moveFocusToFilterCell(toStart?: boolean) {
         const columns = this.grid.filteringService.unpinnedFilterableColumns;
-        if (this.isColumnFullyVisible(columns.length - 1)) {
-            this.grid.filteringService.focusFilterCellChip(columns[columns.length - 1], false);
+        const targetIndex = toStart ? 0 : columns.length - 1;
+        const isVisible = toStart ? this.isColumnLeftFullyVisible(targetIndex) : this.isColumnFullyVisible(targetIndex);
+        if (isVisible) {
+            this.grid.filteringService.focusFilterCellChip(columns[targetIndex], false);
         } else {
-            this.grid.filteringService.scrollToFilterCell(columns[columns.length - 1], false);
+            this.grid.filteringService.scrollToFilterCell(columns[targetIndex], false);
         }
     }
 
+    public navigatePrevFilterCell(column: IgxColumnComponent, eventArgs) {
+        const cols = this.grid.filteringService.unpinnedFilterableColumns;
+        const prevFilterableIndex = this.grid.filteringService.unpinnedFilterableColumns.indexOf(column) - 1;
+        const prevIndex = cols[prevFilterableIndex].visibleIndex;
+        const visibleIndex = column.visibleIndex;
+
+        if (prevIndex >= 0 && visibleIndex > 0 && !this.isColumnLeftFullyVisible(prevIndex) && !column.pinned) {
+            eventArgs.preventDefault();
+            this.grid.filteringService.scrollToFilterCell(this.grid.filteringService.unpinnedFilterableColumns[prevIndex], false);
+        } else if (column.visibleIndex === 0 ||
+                    (prevIndex < 0 && !this.getFirstPinnedFilterableColumn()) ||
+                    column === this.getFirstPinnedFilterableColumn()) {
+            eventArgs.preventDefault();
+        }
+    }
+
+    public navigateNextFilterCell(column: IgxColumnComponent, eventArgs) {
```

Comment:
You can define 'cols' variable like the one in navigatePrevFilterCell method and use it here  wherever it is needed.